lib/group-form/group-form-controller.js

```diff
@@ -0,0 +1,42 @@
+var CONSTANTS = require('../constants');
+
+/**
+ *
+ * A Controller For Editing Group Data.
+ *
+ * @param userMediatorService
+ * @param $stateParams
+ * @param $q
+ * @param $timeout
+ * @param $state
+ *
+ * @constructor
+ */
+function GroupFormController(userMediatorService, $stateParams, $q, $timeout, $state) {
+  var self = this;
+  self.submitted = false;
+
+  var groupId = $stateParams.groupId;
+
+  //If there his no group ID in the URL, we are creating a new group.
+  var readGroupPromise = groupId ? userMediatorService.readGroup(groupId) : $q.when({});
```

Comment:
Looks like this has no sense as if groupId is missing we would just assign null to model. 
We can also move this code to service.